src/Voron/Data/CompactTrees/CompactTree.cs

```diff
@@ -285,6 +304,8 @@ public static CompactTree Create(LowLevelTransaction llt, Tree parent, Slice nam
                     LeafPages = 1,
                     RootPage = newPage.PageNumber,
                     NumberOfEntries = 0,
+                    TreeDictionaryId = dictionaryId,
+                    NextTrainAt = 100000, // We wont try to train the encoder until we have more than 100K entries
```

Comment:
I would rather have a lower value. So we would encounter this more frequently and run this code often.
Maybe 16K, and double from there?

Reply:
Not really the default dictionary should be freaking good (if it's not we are not bootstrapping it properly). The only way a newer dictionary would win is IF it has an out of distribution and we want to delay that until we have a big sample... to say the truth that is like the minimum number I am willing to put there, I have pondered using 1M instead.

Reply:
Code that runs once in a million is a source for bugs, however. I'm fine with making this rare for prod use, but making this more common for debug instances.

Reply:
Good point. For debug we should be talking about the 10000 instead.

Reply:
I wonder if those values shouldn't be configurable from the low level tests. This way we could have more control.